walkstruct.go

```diff
@@ -37,6 +37,34 @@ func doWalkStructElements(t reflect.Type, path []int, f func(reflect.StructField
 	}
 }
 
```

Comment:
```
// WalkStructElementsWithError recursively visits the fields in a structure calling a
// callback for each field.  It modifies the reflect.StructField object
// so that Index is relative to the root object originally passed to
// WalkStructElementsWithError.  This allows the FieldByIndex method on a reflect.Value
// matching the original struct to fetch that field.
//
// WalkStructElementsWithError should be called with a reflect.Type whose Kind() is
// reflect.Struct or whose Kind() is reflect.Ptr and Elem.Type() is reflect.Struct.
// All other types will simply be ignored.
//
// A non-nil return value from the called function stops iteration and recursion and becomes
// the return value.
//
// A special error return value, [DoNotRecurseSignalErr] is not considered an error (it will
// not become the return value, and it does not stop iteration) but it will stop recursion if returned
// on a field that is itself a struct.
```